src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -6,77 +6,291 @@ description: Learn how to add a single Flutter screen to your existing iOS app.
 
-First declare your app delegate to be a subclass of `FlutterAppDelegate`. Then define a FlutterEngine property, which helps you to register a plugin without a FlutterViewController instance.
+{{site.alert.secondary}}
+  The `FlutterEngine` serves as a host to the Dart VM and your Flutter runtime
+  and the `FlutterViewController` attaches to a `FlutterEngine` to pass UIKit
+  input events into Flutter and to display frames rendered by the `FlutterEngine`.
+{{site.alert.end}}
 
-In `AppDelegate.h`:
+The `FlutterEngine` may have the same lifespan as your `FlutterViewController`
+or outlive your `FlutterViewController`.
 
-```objective-c
+{{site.alert.tip}}
+It's generally recommended pre-warm a long-lived `FlutterEngine` for your
+application. This way,
+
+- Flutter screens will render their first frames faster.
+- Your Flutter and Dart state could outlive one `FlutterViewController`.
+- You and your plugins can interact with Flutter and your Dart logic before
+  showing UI.
+{{site.alert.end}}
+
+See [Loading sequence and performance](/docs/development/add-to-app/performance)
+for more analysis on the latency and memory trade-offs of pre-warming an engine.
+
+### Create a FlutterEngine
+
+The proper place to do this is specific to your host app. As an example, we will
+demonstrate creating a `FlutterEngine`, exposed as a property, on app startup in
+the app delegate.
+
+<ul class="nav nav-tabs" id="engine-language" role="tablist">
+  <li class="nav-item">
+    <a class="nav-link active" id="engine-objc" href="#engine-objc-tab" role="tab" aria-controls="engine-objc" aria-selected="true">Objective-C</a>
+  </li>
+  <li class="nav-item">
+    <a class="nav-link" id="engine-swift" href="#engine-swift-tab" role="tab" aria-controls="engine-swift" aria-selected="false">Swift</a>
+  </li>
+</ul>
+
+<div class="tab-content"> {% comment %} FlutterEngine language tab start {% endcomment -%}
+
+<div class="tab-pane active" id="engine-objc-tab" role="tabpanel" aria-labelledby="engine-objc-tab" markdown="1">
+
+**In `AppDelegate.h`:**
+
+<?code-excerpt "AppDelegate.h" title?>
+```objectivec
 @import UIKit;
 @import Flutter;
 
-@interface AppDelegate : FlutterAppDelegate
+@interface AppDelegate : FlutterAppDelegate // More on the FlutterAppDelegate below.
 @property (nonatomic,strong) FlutterEngine *flutterEngine;
 @end
 ```
 
-This allows `AppDelegate.m` to be simple, unless your host app needs to override
-other methods here:
+**In `AppDelegate.m`:**
 
-```objective-c
-#import <FlutterPluginRegistrant/GeneratedPluginRegistrant.h> // Only if you have Flutter Plugins
+<?code-excerpt "AppDelegate.m" title?>
+```objectivec
+#import <FlutterPluginRegistrant/GeneratedPluginRegistrant.h> // Used to connect plugins.
 
 #import "AppDelegate.h"
 
 @implementation AppDelegate
 
-// This override can be omitted if you do not have any Flutter Plugins.
 - (BOOL)application:(UIApplication *)application
     didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
```

Comment:
I know you didn't edit this, but it should be `NSDictionary<UIApplicationLaunchOptionsKey, id> *)`

Reply:
ah thanks